src/blas/impl/KokkosBlas2_gemv_impl.hpp

```diff
@@ -479,6 +479,175 @@ singleLevelGemv (const char trans[],
   }
 }
 
+
+// ---------------------------------------------------------------------------------------------
+// Functor for a two-level parallel_reduce version of (conjugate)
+// transpose GEMV.  The functor uses parallel-for over the columns of the input
+// matrix A and each team uses parallel-reduce over the row of its column.
+// The output vector y is the reduction result.
+template<class AViewType,
+         class XViewType,
+         class YViewType,
+         const bool conj,
+         class IndexType = typename AViewType::size_type>
+struct TwoLevelTransposeGEMV {
+  typedef typename YViewType::non_const_value_type y_value_type;
```

Comment:
Prefer `using new_type = old_type;` C++11 type alias syntax to `typedef old_type new_type;`.